include/wx/event.h

```diff
@@ -2957,14 +2960,32 @@ class WXDLLIMPEXP_CORE wxUpdateUIEvent : public wxCommandEvent
     bool GetSetEnabled() const { return m_setEnabled; }
     bool GetSetShown() const { return m_setShown; }
 
-    void Check(bool check) { m_checked = check; m_setChecked = true; }
+    void Check(bool check) { Set3StateValue(check ? wxCHK_CHECKED : wxCHK_UNCHECKED); }
+    void Set3StateValue(wxCheckBoxState check)
+    {
+        wxCHECK_RET(IsCheckable(), "IsCheckable() not enabled");
+        wxCHECK_RET(Is3State() || check != wxCHK_UNDETERMINED, "Is3State() not enabled");
+        m_3checked = check;
+        m_setChecked = true;
+    }
     void Enable(bool enable) { m_enabled = enable; m_setEnabled = true; }
     void Show(bool show) { m_shown = show; m_setShown = true; }
     void SetText(const wxString& text) { m_text = text; m_setText = true; }
 
     // A flag saying if the item can be checked. True by default.
     bool IsCheckable() const { return m_isCheckable; }
-    void DisallowCheck() { m_isCheckable = false; }
+    void DisallowCheck()
+    {
+        wxCHECK_RET(!GetSetChecked(), "SetCheck() or Set3StateValue() has already been called");
```

Comment:
Why do we need to add this check now when it hasn't been a problem in the past? `DisallowCheck` should only be called in the control code anyway, which should always be before any setting of the check state.

Reply:
FWIW I don't mind adding this assert, although I'd probably use `wxASSERT_MSG()` here too as nothing too bad happens if we don't return. It could be useful for catching logical errors in the library itself.

OTOH we probably should make all these functions with asserts in them non-inline (i.e. move them to `event.cpp`) to avoid having these strings in all call sites.

Reply:
> FWIW I don't mind adding this assert, although I'd probably use `wxASSERT_MSG()` here too as nothing too bad happens if we don't return. It could be useful for catching logical errors in the library itself.
> 
OK

> OTOH we probably should make all these functions with asserts in them non-inline (i.e. move them to `event.cpp`) to avoid having these strings in all call sites.

Good catch!

Reply:
changes force-pushed